src/rt/cast_.d

```diff
@@ -13,6 +13,15 @@
  */
 module rt.cast_;
 
+// because using == does a dynamic cast, but we
+// are trying to implement dynamic cast.
+bool compareClassInfo(ClassInfo a, ClassInfo b)
+{
+    if (a is b)
+        return true;
+    return (a && b) && a.info.name == b.info.name;
```

Comment:
It might be sensible to restrict the name check to `version (Shared)`, i.e., a shared druntime (`.{so,dll}`). I think it's reasonable to expect a process with multiple D binaries to share a single druntime; that's not an option with DMD on Windows, I know.